public/app/features/dashboard/api/publicDashboardApi.ts

```diff
@@ -159,10 +182,16 @@ export const publicDashboardApi = createApi({
         await queryFulfilled;
         dispatch(notifyApp(createSuccessNotification('Public dashboard deleted!')));
 
-        dashboard?.updateMeta({
-          publicDashboardUid: uid,
-          publicDashboardEnabled: false,
-        });
+        if (dashboard instanceof DashboardScene) {
+          dashboard.setState({
+            meta: { ...dashboard.state.meta, publicDashboardUid: uid, publicDashboardEnabled: false },
+          });
+        } else {
+          dashboard?.updateMeta({
+            publicDashboardUid: uid,
+            publicDashboardEnabled: false,
+          });
+        }
```

Comment:
@kaydelaney - Think this is fine, but since the mutations now accept `DashboardModel | DashboardScene` I think we can introduce `updateMeta` method on the `DashboardScene` to avoid the if/else statements

Reply:
that makes sense but I think in that case, you should make both `DashboardModel` and `DashboardScene` implement the same interface so typescript knows the method exists in both classes?
I'm not sure if you just declare both methods in both classes that would work

Reply:
> I'm not sure if you just declare both methods in both classes that would work

yes, this should work @juanicabanas